### PR TITLE
op-node: Audit use of L1 State

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

The L1 State can return uninitialized values. I have checked every usage in the op-node and either check the return value from L1 state if is important or let the empty value through.

**Additional context**

This is a second approach to solve the problem that #3952 attempts to solve. After thinking about it a bit longer, I think that this solution works a bit better.

**Metadata**
- Fixes ENG-2920

### CODE REVIEW

File: op-node/rollup/driver/conf_depth.go [hunk 1]
```diff
@@ -33,7 +32,10 @@ func (c *confDepth) L1BlockRefByNumber(ctx context.Context, num uint64) (eth.L1B
 
 	// Don't apply the conf depth is l1Head is empty (as it is during the startup case before the l1State is initialized).
 	l1Head := c.l1Head()
-	if num == 0 || c.depth == 0 || num+c.depth <= l1Head.Number || l1Head.Hash == (common.Hash{}) {
+	if l1Head == (eth.L1BlockRef{}) {
```
Comment (tynes) [inline op-node/rollup/driver/conf_depth.go:35]: Are these 2 cases just split up for readability?

Comment (trianglesphere) [reply op-node/rollup/driver/conf_depth.go:35]: yea

**Review by tynes — APPROVED** [review]

**Review by protolambda — APPROVED** [review]

**Comment by tynes** [comment]
@trianglesphere What is the deal with this PR? Still trying to get it merged?

**Review by tynes — APPROVED** [review]

**Comment by trianglesphere** [comment]
@mergify refresh

**Pull request merged**